### ISSUE TITLE
No clear error if authorization token has been expired

### ISSUE THREAD

**Issue by datakurre** [collective/gatsby-source-plone#78]
It turned out that Plone REST API authorization tokens expire in 12 hours by default.

Expired token can only be seen currently as content not found:

```
error gatsby-node.js returned an error
  TypeError: Cannot read property 'allPloneDocument' of undefined
```

* [ ] better error message if authorization token has been expired
* [ ] document renewing token manually (special renew endpoint works only before token has been already expired)

Nice thing would be to get custom timeout support for plone.restapi login endpoint and instruct to create token with longer expiration.

Shorter expiration is more safe, of course, but not practical for use-case where CI builds gatsby site from Plone.  /cc @tisto @sneridagh

**Comment by ajayns** [comment]
If plone.restapi itself leaves only such a vague message as an error, it won't be possible in the plugin to handle and display a better error message right?

**Comment by cekk** [comment]
You have this error also if no token is set (it tooks me 15minutes to understand what was the problem).
Could be a good idea to make some checks before calling API and provide an useful error message to the user.

**Comment by tisto** [comment]
@ajayns what kind of message would you expect from plone.restapi? Could you describe this in detail in an issue (with an example of the current response and the response you would expect). Error reporting is certainly something that we have to improve for plone.restapi.

**Comment by datakurre** [comment]
@tisto Dedicated endpoint for checking validity of token with valid:true/false response could be enough.

**Comment by ajayns** [comment]
Yes that could help us handle what message to show in the case of a token expired. Also, is it possible in any way to extend the validity of the token from the current 12 hour default?

**Comment by datakurre** [comment]
Related: while learning gatsby cache usage I noticed that some plugins are using "plugin state" to renew their access tokens so that next build has always a fresh token. That would be interesting once Plone REST API has better API for token management.

**Comment by iFlameing** [comment]
I also faced the same problem during the build process of gatsby-source-plugin and after lot of research I found that if we don't provide the data which is needed by graphql query or we added the data in private mode and don't provide the updated the token, we get above error message.I tried with both with token and without token if  graphql queries  is satisfied by the published state then this error will not rise.
